Core/include/Acts/Utilities/Result.hpp

```diff
@@ -172,6 +175,144 @@ class Result {
     return std::move(std::get<T>(m_var));
   }
 
+  /// Retrieves the valid value from the result object, or returns a default
+  /// value if no valid value exists.
+  ///
+  /// @param[in] v The default value to use if no valid value exists.
+  /// @note This is the lvalue version.
+  /// @note This function always returns by value.
+  /// @return Either the valid value, or the given substitute.
+  template <typename U>
+  T value_or(U&& v) const&
+    requires(std::same_as<std::decay_t<U>, T>)
+  {
+    if (ok()) {
+      return value();
+    } else {
+      return std::forward<U>(v);
+    }
+  }
+
+  /// Retrieves the valid value from the result object, or returns a default
+  /// value if no valid value exists.
+  ///
+  /// @param[in] v The default value to use if no valid value exists.
+  /// @note This is the rvalue version which moves the value out.
+  /// @note This function always returns by value.
+  /// @return Either the valid value, or the given substitute.
+  template <typename U>
+  T value_or(U&& v) &&
+    requires(std::same_as<std::decay_t<U>, T>)
+  {
+    if (ok()) {
+      return std::move(*this).value();
+    } else {
+      return std::forward<U>(v);
+    }
+  }
+
+  /// Transforms the value contained in this result.
+  ///
+  /// Applying a function `f` to a valid value `x` returns `f(x)`, while
+  /// applying `f` to an invalid value returns another invalid value.
+  ///
+  /// @param[in] callable The transformation function to apply.
+  /// @note This is the lvalue version.
+  /// @note This functions is `fmap` on the functor in `A` of `Result<A, E>`.
+  /// @return The modified valid value if exists, or an error otherwise.
+  template <typename C>
+  auto transform(C&& callable) const&
+    requires std::invocable<C, const T&>
+  {
+    using R =
+        Result<std::decay_t<decltype(std::declval<C>()(std::declval<T>()))>, E>;
```

Comment:
this hurts my brain - but I guess there is no way around it?

Reply:
maybe lambdas in unevaluated contexts could help here, but I don't think its that bad

Reply:
This is now two smaller typedefs, hopefully aiding readability.